File: 0905-sort-array-by-parity/0905-sort-array-by-parity.py
class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        ans=[0]*len(nums)
        start=0
        end=len(nums)-1
        for i in nums:
            if i%2==0:
                ans[start]=i
                start+=1
            else:
                ans[end]=i
                end-=1
        return ans

File: 0905-sort-array-by-parity/0905-sort-array-by-parity.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int size = nums.length;
        int[] newArray = new int[size];
        for (int i = 0; i < size; i++) {
            newArray[i] = 0;
        }
        int start=0;
        int end=size-1;
        for(int i=0;i<size;i++){
            if(nums[i]%2==0){
                newArray[start]=nums[i];
                start+=1;
            }
            else{
                newArray[end]=nums[i];
                end-=1;
            }
        }
        return newArray;
    }
}
